### ISSUE TITLE
CA2119 false positive

### ISSUE THREAD

**Issue by avivanoff** [dotnet/roslyn-analyzers#4566]
### Analyzer

**Diagnostic ID**: [https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2119): `Seal methods that satisfy private interfaces`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 5.0.1 (Latest)

### Describe the bug

See attached project.

### Steps To Reproduce

Load and compile the attached project.

### Expected behavior

CA2119 should not fire.

### Actual behavior

CA2119 fires.

## Additional context

It would be beneficial if CA2119 reported which interfaces/methods caused the rule violation.

[CA2119.ZIP](https://github.com/dotnet/roslyn-analyzers/files/5682491/CA2119.ZIP)

**Comment by Youssef1313** [comment]
Have you tried to make the Form `sealed` or `internal`?

**Comment by avivanoff** [comment]
> Have you tried to make the Form `sealed` or `internal`?

Yes, making the form sealed or internal fixes the error, but I am building an extendable form library, so this is not an option.

**Comment by Evangelink** [comment]
@avivanoff I don't have any access to DevExpress so I can't compile your code, could you please give us some more info about the issue?

**Comment by avivanoff** [comment]
@Evangelink, is there a way to privately send you the project, that does not require access to DevExpress NuGet feed?

**Comment by Evangelink** [comment]
Yes sure you can send me an email but do note that I am NOT working at Microsoft, only helping on my free time. I honestly won't do anything with your app but I'd rather have you informed beforehand.

**Comment by avivanoff** [comment]
@Evangelink, I might be missing something, but where do I find your email?

**Comment by Evangelink** [comment]
amaury[dot]leve[at]gmail[dot]com

**Comment by avivanoff** [comment]
@Evangelink, email sent.

**Comment by Evangelink** [comment]
Thank you @avivanoff! I was able to track down the issue and I confirm there is a false positive with current implementation.

I will provide a fix soon, in the meantime feel free to ignore this issue.

Minimal repro:

```csharp
public class C : System.Windows.Forms.Control {} // diagnostic - Control implements IArrangedElement
```

The issue is due to the fact we loop through all interfaces instead of looking only at direct interfaces.

**Comment by Evangelink** [comment]
Woops looks like I did introduce this issue when fixing #4406, I am now scanning ALL interfaces instead of current interfaces and all their interfaces.
